### PR TITLE
Add support for translating DateTimeOffset.LocalDateTime and DateTimeOffset.UtcDateTime

### PR DESCRIPTION
**Pull request by leotsarev**
Undecided: Should I do something else for other providers? SqlServer.Compact

### CODE REVIEW

**Comment by ErikEJ** [comment]
SQL Server Compact only has the datetime data type, so nothing to do there

File: src/EntityFramework/Core/Common/CommandTrees/ExpressionBuilder/EdmFunctions.cs [hunk 1]
```diff
@@ -481,6 +481,28 @@ public static DbFunctionExpression GetTotalOffsetMinutes(this DbExpression dateT
             return InvokeCanonicalFunction("GetTotalOffsetMinutes", dateTimeOffsetArgument);
         }
 
+        /// <summary>
+        /// Creates a <see cref="T:System.Data.Entity.Core.Common.CommandTrees.DbFunctionExpression" /> that gets local DateTime using CAST expressions from specified argument, which must have a DateTimeOffset result type. The result type of the expression is DateTime.
+        /// </summary>
+        /// <returns>A new DbFunctionExpression that cast DATETIMEOFFSET to DATEIME2, discarding offset part</returns>
```
Comment (ErikEJ) [inline src/EntityFramework/Core/Common/CommandTrees/ExpressionBuilder/EdmFunctions.cs]: Small typo

File: src/EntityFramework/Core/Common/CommandTrees/ExpressionBuilder/EdmFunctions.cs [hunk 2]
```diff
@@ -481,6 +481,28 @@ public static DbFunctionExpression GetTotalOffsetMinutes(this DbExpression dateT
...
+        public static DbFunctionExpression LocalDateTime(this DbExpression dateTimeOffsetArgument)
+        {
+            Check.NotNull(dateTimeOffsetArgument, "dateTimeOffsetArgument");
+            return InvokeCanonicalFunction("LocalDateTime", dateTimeOffsetArgument);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="T:System.Data.Entity.Core.Common.CommandTrees.DbFunctionExpression" /> that gets UTC DateTime using CAST expressions from specified argument, which must have a DateTimeOffset result type. The result type of the expression is DateTime.
+        /// </summary>
+        /// <returns>A new DbFunctionExpression that cast DATETIMEOFFSET to Utc DATEIME2 </returns>
```
Comment (ErikEJ) [inline src/EntityFramework/Core/Common/CommandTrees/ExpressionBuilder/EdmFunctions.cs]: Small typo

**Comment by leotsarev** [comment]
@ErikEJ sorry for my English. Did I fix correct thing?

**Comment by ErikEJ** [comment]
I actually meant: DATEIME2 > DATETIME2

**Comment by leotsarev** [comment]
@ErikEJ oops thanks

**Comment by ajcvickers** [comment]
Merged. Thanks!

**Pull request closed without merging**